src/client/Microsoft.Identity.Client.Broker/RuntimeBroker.cs

```diff
@@ -440,7 +440,7 @@ public bool IsBrokerInstalledAndInvokable(AuthorityType authorityType)
                 _logger.Info("[WAM Broker] MsalRuntime init failed...");
                 _logger.InfoPii(s_initException);
 
-                return false;
```

Comment:
This is not the correct fix I don't think. We want to propagate all exceptions except the ApiContractViolation which states that msal_runtime cannot be initialized (e.g. we're on Win8).

I would suggest instead to update the initialization method like this:


```csharp
      private static Lazy<NativeInterop.Core> s_lazyCore = new Lazy<NativeInterop.Core>(() =>
        {
            try
            {
                AppDomain.CurrentDomain.ProcessExit += OnProcessExit;

                return new NativeInterop.Core();
            }                  
            catch (MsalRuntimeException ex) when (ex.Status == ResponseStatus.ApiContractViolation)
            {
                // failed to initialize msal runtime - can happen on older versions of Windows. Means broker is not available.
                s_initException = ex;

                // ignored
                return null;
            }

            // any other exceptions are not expected and should bubble up to the app developer, so that they can log bugs.
        });

```

Reply:
(if this is acceptable, consider it as sign off from me).

Reply:
agreed, did not think of `MsalRuntimeException`